libraries/chain/transaction.cpp

```diff
@@ -343,10 +344,35 @@ signed_transaction packed_transaction::get_signed_transaction() const
 
 }
 
-void packed_transaction::set_transaction(const transaction& t, packed_transaction::compression_type _compression)
+packed_transaction::packed_transaction( bytes&& packed_txn, vector<signature_type>&& sigs,
+                                        bytes&& packed_cfd, vector<bytes>&& cfd, compression_type _compression )
+:signatures(std::move(sigs))
+,compression(_compression)
+,packed_context_free_data(std::move(packed_cfd))
+,packed_trx(std::move(packed_txn))
+{
+   EOS_ASSERT(packed_cfd.empty() || cfd.empty(), tx_decompression_error, "Invalid packed_transaction");
+   if( !cfd.empty() ) {
+      set_context_free_data(cfd);
+   }
+}
+
+packed_transaction::packed_transaction( signed_transaction&& t, bytes&& packed_cfd, compression_type _compression )
```

Comment:
I think it would be clearer if this constructor was split into two:

- `packed_transaction( signed_transaction&& t, compression_type _compression )`
- `packed_transaction( transaction&& t, vector<signature_type>&& sigs, bytes&& packed_cfd, compression_type _compression )`

Reply:
Fixed in #6494